### PR TITLE
`PyScope`/`PyModule` API cleanup

### PR DESCRIPTION
**Pull request by lostmsu**
- removed `PyScopeManager`
- merged `PyScope` into `PyModule` (every `PyScope` was a Python module anyway)
- minor behavioral changes

### Does this close any currently open issues?

This is part of [3.0 API cleanup](https://github.com/pythonnet/pythonnet/issues/1539), targeting https://github.com/pythonnet/pythonnet/issues/1390

### Checklist

-   [x] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by lostmsu** [comment]
@filmor can you take a look at this one at least in principle?

File: src/runtime/module.cs [hunk 1]
```diff
@@ -101,26 +151,15 @@ public PyScope NewScope()
         /// Import a scope or a module of given name,
         /// scope will be looked up first.
```
Comment (filmor) [inline src/runtime/module.cs:152]: This part of the comment is not true anymore.

**Comment by filmor** [comment]
I'm okay with this one, though for a slight backwards-compatibility I'd propose we add `[Obsolete]public class PyScope : PyModule` with both public constructors copied. Is there any particular reason why you renamed the file from `pymodule` to `module`? Matching the filename makes sense to me.

**Review by filmor — APPROVED** [review]

**Comment by lostmsu** [comment]
@filmor re: filename. Without renaming the file, diff is larger (that's because the file has more code from `PyScope` than from `PyModule`)

**Comment by lostmsu** [comment]
@filmor re: `PyScope`. The original `PyScope` did not have any public constructors.

**Comment by filmor** [comment]
It had public static functions, though (and that's also how I always used it). So you can probably ignore the constructors, but deriving still makes sense.

**Comment by lostmsu** [comment]
@filmor I don't see any public static functions on old `PyScope`. There was a single factory method `Py.Scope()` and I preserved that one.

**Pull request merged**